futures-util/src/try_future/mod.rs

```diff
@@ -51,19 +51,42 @@ pub use self::unwrap_or_else::UnwrapOrElse;
 mod try_chain;
 crate use self::try_chain::{TryChain, TryChainAction};
 
-impl<F: TryFuture> TryFutureExt for F {}
+impl<Fut: TryFuture> TryFutureExt for Fut {}
 
-/// Adapters specific to `Result`-returning futures
+/// Adapters specific to [`Result`]-returning futures
 pub trait TryFutureExt: TryFuture {
-    /// Flatten the execution of this future when the successful result of this
-    /// future is a sink.
+    /// Flattens the execution of this future when the successful result of this
+    /// future is a [`Sink`].
     ///
     /// This can be useful when sink initialization is deferred, and it is
     /// convenient to work with that sink as if sink was available at the
     /// call site.
     ///
     /// Note that this function consumes this future and returns a wrapped
     /// version of it.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// #![feature(futures_api)]
+    /// use futures::future::{Future, TryFutureExt};
+    /// use futures::sink::Sink;
+    /// # use futures::channel::mpsc::{self, SendError};
+    /// # type T = i32;
+    /// # type E = SendError;
+    ///
+    /// fn make_sink_async() -> impl Future<Output = Result<
+    ///     impl Sink<SinkItem = T, SinkError = E>,
+    ///     E,
+    /// >> { // ... }
+    /// # let (tx, _rx) = mpsc::unbounded::<i32>();
```

Comment:
Just realised we’re missing an equivalent of [`io::Sink`](https://doc.rust-lang.org/nightly/std/io/struct.Sink.html) for `Sink`s (unfortunate naming collision though). I might submit a PR for one tomorrow.

Reply:
Please don't call it `Sink` or `Sync` XD